srcs/app/server/FDsets.cpp

```diff
@@ -6,20 +6,33 @@ FDsets::FDsets() {
 	FD_ZERO(&write_set_);
 }
 
-void	FDsets::removeFromReadSet(int fd) {
-	FD_CLR(fd, &read_set_);
+void	FDsets::addToWriteSet(int fd) {
+	FD_SET(fd, &write_set_);
+}
+
+void	FDsets::addToReadSet(int fd) {
+	FD_SET(fd, &read_set_);
+	if (max_fd_ < fd) {
+		max_fd_ = fd;
+	}
 }
 
 void	FDsets::removeFromWriteSet(int fd) {
 	FD_CLR(fd, &write_set_);
 }
 
-void	FDsets::addToWriteSet(int fd) {
-	FD_SET(fd, &write_set_);
+void	FDsets::removeFd(int fd) {
+	removeFromReadSet_(fd);
+	removeFromWriteSet(fd);
+	if (fd == max_fd_) {
+		while (!isReadSet(max_fd_)) {
```

Comment:
`isReadSet()` evaluates the temporary set (the one which `select()` is modifiying).

```suggestion
		while (!FD_ISSET(max_fd_, &read_set_) && !FD_ISSET(max_fd_, &write_set_)) {
```